### DISCUSSION TITLE
Named Stores

### DISCUSSION THREAD

**Discussion by litvinenkow** [vuejs/pinia#2342] [Ideas]
Don't you thought of named Stores like named Routes?
You have just maybe 3-5 similar entities that uses same stores but some internal options that changes inside options within name change and you need to setup them one for one, why not just create some store like
export const useNamedStore = defineNamedStore('namedstore', 'myneedednameorparam', {...options});
and use it like 
const myNamedStore = useNamedStore('myneedednameorparam');
maybe with some internal access to this "nameorparam" inside

**Comment by posva** [comment]
sorry for the late reply. This has been discussed a couple of times, the idea of creating multiple instances of a store goes against the very same principle of what a store is. In most scenarios, you just need a composable with some cache mechanism
